.github/scripts/parse_rpt.py

```diff
@@ -1,33 +1,62 @@
 import sys
+import json
+import os
+import re, subprocess
+sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
+from common.check_gen_files import check_gen_files
 
 sys.stdout.flush()
 
-if len(sys.argv) == 1 or sys.argv[1] == "sky130hvl_ldo":
+dir_path = r'flow/reports'
+lib = os.listdir(dir_path)
+
+if (len(sys.argv) > 1) and (sys.argv[1] == str(lib[0])):
+    drc_filename = "flow/reports/" + sys.argv[1] + "/cryo/6_final_drc.rpt"
+    lvs_filename = "flow/reports/" + sys.argv[1] + "/cryo/6_final_lvs.rpt"
+elif (len(sys.argv) == 1) or (sys.argv[1] == "sky130hvl_ldo"):
     drc_filename = "work/6_final_drc.rpt"
     lvs_filename = "work/6_final_lvs.rpt"
 else:
     drc_filename = "work/"+sys.argv[1]+"/6_final_drc.rpt"
     lvs_filename = "work/"+sys.argv[1]+"/6_final_lvs.rpt"
 
-if len(sys.argv) > 1 and sys.argv[1] == "sky130hvl_ldo":
+if (len(sys.argv) > 1) and (sys.argv[1] == "sky130hvl_ldo"):
     with open(drc_filename, 'r') as f1, open("../../../.github/scripts/expected_drc_reports/expected_ldo_drc.rpt", 'r') as f2:
         content1 = f1.readlines()
         content2 = f2.readlines()
-        if content1 == content1:
+        if content1 == content2:
             print("DRC is clean!")
         else:
             raise ValueError("DRC failed!")
-        
 elif sum(1 for line in open(drc_filename)) > 3:
     raise ValueError("DRC failed!")
 else:
     print("DRC is clean!")
 
-
-with open(lvs_filename) as f:
-    f1 = f.read()
-
-    if "failed" in f1:
+# cryo LVS check 
+if (len(sys.argv) > 1) and (sys.argv[1] == str(lib[0])):
```

Comment:
@chetanyagoyal : can you explain more about this please? what is the difference between the checks that is done in this `if` block and the `else` block. looks like they both are using the same `lvs_filename`?

Reply:
I think this one is required because the cryo's parse report file checks the drc and lvs in `flow/reports`. Seems like the work directory doesn't exist for it, which is why i added the if statement

the `lib = os.listdir(..)` statement checks which cryo library was used to go one level deeper into `flow/reports`

Reply:
that's alright but you are checking the file with the same variable. the code under `if` and `else` logically does the same thing ?

Reply:
Yes i've fixed this now 
If checks for temp-sense and ldo, and the elif checks for the various possible cryo-gens